src/store/helpers.js

```diff
@@ -26,3 +28,111 @@ export function onlyHandleAPIError (error, handleFn) {
     throw error
   }
 }
+
+/**
+ * Defines a vues module that can be used to handle async request metadata.
+ *
+ * Instantiation:
+
+ const meta = defineRequestModule()
+ export const modules = { meta }
+
+ * Usage:
+
+async save ({ commit, dispatch }, group) {
+  dispatch('meta/request', {
+    id: `save-${group.id}`,
+    async run () {
+      const updatedGroup = await groups.save(group)
+      commit(types.RECEIVE_GROUP, { group: updatedGroup }) // other commits when successful
+    },
+  })
+},
+ */
+export function defineRequestModule () {
+  const types = {
+    REQUEST: 'Request',
+    RECEIVE_SUCCESS: 'Receive Success',
+    RECEIVE_ERROR: 'Receive Error',
+    CLEAR: 'Clear',
+  }
+
+  const state = {
+    entries: {},
+  }
+
+  const getters = {
+    get: state => id => state.entries[id] || ({
+      isWaiting: false,
+      error: null,
+      success: false,
+    }),
+    error: (state, getters) => id => field => {
+      const error = getters.get(id).error
+      return error && error[field] && error[field][0]
+    },
+    isWaiting: (state, getters) => id => getters.get(id).isWaiting,
+    success: (state, getters) => id => getters.get(id).success,
+  }
+
+  const actions = {
+    async request ({ commit }, { id, run }) {
+      commit(types.REQUEST, { id })
+      try {
+        await run()
+        commit(types.RECEIVE_SUCCESS, { id })
+      }
+      catch (error) {
+        const { response: { status = -1, data } = {} } = error
+        if (status >= 400 && status < 500) {
```

Comment:
This would still need to split the errors that the user can handle (validation errors) from those that they can't (permission errors).

Would be nice to extract that logic out to a helper, so we can test it well. A few standalone functions should suffice, something like:
- `isValidationError (error) => boolean` (need to show validation on page)
- `isRequestError (error) => boolean` (if not also a validation error, means it was handled by axios interceptor and a toast was displayed)

.. if neither of those, then it's some other error.

Reply:
Extracting the checks seems like a good idea. Though we would need still need the additional logic to get the response data, so the code won't get shorter, only more explicit.